src/components/DeleteModalContainer.tsx

```diff
@@ -0,0 +1,50 @@
+import { Dispatch, RefObject, SetStateAction } from 'react';
+import { Task } from '../utils/types';
+import { deleteAllTasks } from '../utils/api';
+import DeleteModalUI from './DeleteModalUI';
+
+interface DeleteModalContainerProps {
+  deleteModalRef: RefObject<HTMLDialogElement>;
+  tasks: Task[];
+  setTasks: Dispatch<SetStateAction<Task[]>>;
```

Comment:
🤔 Question: same thing I mentioned above; can we make this a generic function instead? You're tying the component to your implementation quite a bit here.

Reply:
updated to use generic `setUpdateFunction`